spec/solver_spec.rb

```diff
@@ -0,0 +1,49 @@
+require_relative '../solver'
+
+describe Solver do
+  before :all do
+    @solver = Solver.new
+  end
+
+  context 'factorial method' do
+    it 'should compute factorial for non-zero positive number' do
+      expect(@solver.factorial(5)).to eq(120)
+    end
+
+    it 'should return 1 for number 0' do
+      expect(@solver.factorial(0)).to eq(1)
+    end
```

Comment:
- [ ] Special case when input is 0, the result should equal 1
![image](https://user-images.githubusercontent.com/62705947/218067872-f0da8192-b0e7-4d7a-bd04-5ab94440f70e.png)

![image](https://user-images.githubusercontent.com/62705947/218067789-c58fff7e-c201-4fc3-8372-b907592df86b.png)

Reply:
Hi @GzimAsani, the `method` is already returning 1, when the input is 0. Can please give more details?

Reply:
@Abdessittir, you are correct, the method returns 1 when 0 is inputted. Great work 😎